From e4d197971717ae14d5d9062e87ba22fbc43a28ca Mon Sep 17 00:00:00 2001
From: Vincent Becker <vincentx.becker@intel.com>
Date: Thu, 13 Dec 2012 19:49:57 +0100
Subject: [PATCH 39/49] [AUDIO] Add implementation of FM Rx volume check and
 set

BZ: 74833

Add FM Rx stream support for volume check and set, by
calling the dedicated audio HAL interface.

Change-Id: I0d46f314b4a1075c54e7336d25267aa00f0e7289
Signed-off-by: Vincent Becker <vincentx.becker@intel.com>
Reviewed-on: http://android.intel.com:8080/81486
Reviewed-by: Dixon, CharlesX <charlesx.dixon@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 66219c1..9da5304 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -3161,6 +3161,12 @@ status_t AudioPolicyManagerBase::checkAndSetVolume(int stream,
         if (stream == AudioSystem::BLUETOOTH_SCO) {
             mpClientInterface->setStreamVolume(AudioSystem::VOICE_CALL, volume, output, delayMs);
         }
+
+        // Update FM Rx volume
+        if (stream == AudioSystem::FM_RX) {
+            return mpClientInterface->setFmRxVolume(volume, delayMs);
+        }
+
         mpClientInterface->setStreamVolume((AudioSystem::stream_type)stream, volume, output, delayMs);
     }
 
-- 
1.7.9

